recipe/build-mpi.sh

```diff
@@ -33,7 +33,29 @@ if [[ "$target_platform" == osx-* ]]; then
   # Add gfortran internal header to clang include dir
   fcdir=$($FC -print-search-dirs | awk '/install: /{print $2}')
   ccdir=$($CC -print-search-dirs | awk '/libraries: =/{print substr($2,2)}')
-  ln -s $fcdir/include/ISO_Fortran_binding.h $ccdir/include
+  echo "=== DEBUG ==="
+  echo "cp ${fcdir}include/ISO_Fortran_binding.h ${ccdir}include/ISO_Fortran_binding.h"
+  cp ${fcdir}include/ISO_Fortran_binding.h ${ccdir}include/ISO_Fortran_binding.h
```

Comment:
You are missing a slash here. To be sure of getting it right just do

```suggestion
  echo "cp ${fcdir}/include/ISO_Fortran_binding.h ${ccdir}/include"
  cp ${fcdir}/include/ISO_Fortran_binding.h ${ccdir}/include
```

Reply:
Unfortunately, that won't help. (It was that way at the beginning, I've just removed it, to make sure, that it is not the double slash before include, which causes problems.)

If you look at the CI with the current build-mpi.sh, you see
```
2024-01-31T16:39:08.2427120Z cp /Users/runner/miniforge3/conda-bld/mpich-mpi_1706718393956/_build_env/lib/gcc/arm64-apple-darwin20.0.0/12.3.0/include/ISO_Fortran_binding.h /Users/runner/miniforge3/conda-bld/mpich-mpi_1706718393956/_build_env/lib/clang/16include/ISO_Fortran_binding.h
2024-01-31T16:39:08.2471840Z cp: /Users/runner/miniforge3/conda-bld/mpich-mpi_1706718393956/_build_env/lib/gcc/arm64-apple-darwin20.0.0/12.3.0/include/ISO_Fortran_binding.h: No such file or directory
```

The file is simply missing.

When you install the package manually, do you get GFortran 12 or 13 on your osx_arm64 platform. According to the CI output, GFortran 12.3 is used!